ansible_collections/arista/avd/roles/eos_designs/docs/tables/isis-settings.md

```diff
@@ -49,6 +51,10 @@
       # Local convergence delay in milliseconds.
       local_convergence_delay: <int; default=10000>
 
+    # Type-7 encrypted password.
+    underlay_isis_authentication_key: <str>
+    underlay_isis_authentication_mode: <str; "md5" | "text">
```

Comment:
p2p_links inherits this setting and this schema has only these two types defined. May be we need to refactor this later.

Reply:
@ClausHolbechArista Shall we add all modes in schema.
isis authentication in p2p_links: https://github.com/aristanetworks/avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_p2p_links.schema.yml#L89

Reply:
Changing it would require a larger model to cater for all the isis auth options. We added them all in eos_cli_config_gen recently, so if that model could be reused it would be great. But I don't see how without breaking what is there.

Reply:
Added sha, shared secret support on interface level. To support this at host level, Shall I add keys  for sha, shared-secret parameters on host level. For now added support for md5/text on host level.

Reply:
As mentioned in the other comment, I think this PR is going too far outside what was requested. And it is not tackling all of what was requested (you also need to honor the underlay_ setting in the underlay code. Last I checked you did only look at this for l3_edge/core_interfaces.